tsdb/store.go

```diff
@@ -1372,13 +1372,44 @@ func (s *Store) DeleteSeriesWithPredicate(ctx context.Context, database string,
 			return err
 		}
 
+		measurementName := make([]byte, 0)
+
+		if measurement != nil {
+			if m, ok := measurement.(*influxql.BinaryExpr); ok {
+				rhs, ok := m.RHS.(*influxql.VarRef)
+				if ok {
+					measurementName = []byte(rhs.Val)
+					exists, err := sh.MeasurementExists(measurementName)
+					if err != nil {
+						return err
+					}
+					if !exists {
+						return nil
+					}
+				}
+			}
+		}
+
 		// Find matching series keys for each measurement.
 		mitr, err := index.MeasurementIterator()
 		if err != nil {
 			return err
 		}
 		defer mitr.Close()
 
+		deleteSeries := func(mm []byte) error {
```

Comment:
Can we capture the error from `mitr.Close()` with Capture() call?  It is not new code, but a good thing to add while the hood is open.